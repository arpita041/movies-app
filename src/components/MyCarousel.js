import React from 'react'
import { Navbar } from 'react-bootstrap'
import '../css/MyCarousel.scss'
import NavBar from './NavBar'
function MyCarousel() {
    return (
        <div>
            <NavBar></NavBar>
            <div className='content content--margin'>
                <div className="carousel" aria-label="Gallery">
                    <ol className="carousel__viewport">
                        <li id="carousel__slide1"
                            className="carousel__slide">
                            <div className="carousel__snapper">
                                <img width='100%' height='100%' src="./288732.jpg" alt="" />
                            </div>
                            <a href="#carousel__slide4"
                                className="carousel__prev">Go to last slide</a>
                            <a href="#carousel__slide2"
                                className="carousel__next">Go to next slide</a>

                        </li>
                        <li id="carousel__slide2"
                            className="carousel__slide">
                            <div className="carousel__snapper">
                                <img width='100%' height='100%' src="./329583.jpg" alt="" />
                            </div>
                            <a href="#carousel__slide1"
                                className="carousel__prev">Go to previous slide</a>
                            <a href="#carousel__slide3"
                                className="carousel__next">Go to next slide</a>
                        </li>
                        <li id="carousel__slide3"
                            className="carousel__slide">
                            <div className="carousel__snapper">
                                <img width='100%' height='100%' src="./790933.jpg" alt="" />

                            </div>
                            <a href="#carousel__slide2"
                                className="carousel__prev">Go to previous slide</a>
                            <a href="#carousel__slide4"
                                className="carousel__next">Go to next slide</a>
                        </li>
                    </ol>
                    <aside className="carousel__navigation">
                        <ol className="carousel__navigation-list">
                            <li className="carousel__navigation-item">
                                <a href="#carousel__slide1"
                                    className="carousel__navigation-button">Go to slide 1</a>
                            </li>
                            <li className="carousel__navigation-item">
                                <a href="#carousel__slide2"
                                    className="carousel__navigation-button">Go to slide 2</a>
                            </li>
                            <li className="carousel__navigation-item">
                                <a href="#carousel__slide3"
                                    className="carousel__navigation-button">Go to slide 3</a>
                            </li>
                        </ol>
                    </aside>
                </div>
            </div>

        </div>
    )
}

export default MyCarousel
