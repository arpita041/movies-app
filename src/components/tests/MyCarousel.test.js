import React from 'react';
import { render, screen } from '@testing-library/react';
import MyCarousel from '../MyCarousel';
import { BrowserRouter } from 'react-router-dom';



test('to check whether AllDirectors component rendered',()=>{
    render(
        <BrowserRouter>
        <MyCarousel />
    </BrowserRouter>);
    
})

// asFragment(<App />) matches toMatchSnapshot() (the matcher provided by jest-dom).
test('should take a snapshot', () => {
    const { asFragment } = render(
        <BrowserRouter>
        <MyCarousel />
    </BrowserRouter>);
    
    expect(asFragment(  
    <BrowserRouter>
        <AllDirectors />
    </BrowserRouter>)).toMatchSnapshot()
   })