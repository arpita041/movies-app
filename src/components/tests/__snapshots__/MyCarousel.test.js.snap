// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="nav"
      >
        <input
          id="nav-check"
          type="checkbox"
        />
        <div
          class="nav__header"
        >
          <div
            class="nav__title"
          >
            <a
              class="nav__link--color"
              href="/home"
            >
              CinemaWall
            </a>
          </div>
        </div>
        <div
          class="nav__btn"
        >
          <label
            for="nav-check"
          >
            <span>
              ...
            </span>
            <span>
              ...
            </span>
            <span>
              ...
            </span>
          </label>
        </div>
        <div
          class="nav__links"
        >
          <div
            class="nav__link"
          >
            <a
              class="nav__link--color"
              href="/forms"
            >
              Add Cinema
            </a>
             
          </div>
          <div
            class="nav__link"
          >
            <a
              class="nav__link--color"
              href="/addDirector"
            >
              Add Director
            </a>
          </div>
          <div
            class="nav__link"
          >
            <a
              class="nav__link--color"
              href="/showMovies"
            >
              Movie List
            </a>
          </div>
          <div
            class="nav__link"
          >
            <a
              class="nav__link--color"
              href="/director"
            >
              Director List
            </a>
          </div>
          <div
            class="nav__link"
          >
            <div
              class="dropdown"
            >
              <button
                class="nav__dropbtn"
              >
                Other Options
              </button>
              <div
                class="dropdown-content"
              >
                <span
                  class="dropdown__link"
                >
                  <a
                    class="nav__link--color"
                    href="/deleteMovie"
                  >
                    Delete a Movie
                  </a>
                </span>
                <span
                  class="dropdown__link"
                >
                  <a
                    class="nav__link--color"
                    href="/update"
                  >
                    Update Details
                  </a>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="content content--margin"
    >
      <div
        aria-label="Gallery"
        class="carousel"
      >
        <ol
          class="carousel__viewport"
        >
          <li
            class="carousel__slide"
            id="carousel__slide1"
          >
            <div
              class="carousel__snapper"
            >
              <img
                alt=""
                height="100%"
                src="./288732.jpg"
                width="100%"
              />
            </div>
            <a
              class="carousel__prev"
              href="#carousel__slide4"
            >
              Go to last slide
            </a>
            <a
              class="carousel__next"
              href="#carousel__slide2"
            >
              Go to next slide
            </a>
          </li>
          <li
            class="carousel__slide"
            id="carousel__slide2"
          >
            <div
              class="carousel__snapper"
            >
              <img
                alt=""
                height="100%"
                src="./329583.jpg"
                width="100%"
              />
            </div>
            <a
              class="carousel__prev"
              href="#carousel__slide1"
            >
              Go to previous slide
            </a>
            <a
              class="carousel__next"
              href="#carousel__slide3"
            >
              Go to next slide
            </a>
          </li>
          <li
            class="carousel__slide"
            id="carousel__slide3"
          >
            <div
              class="carousel__snapper"
            >
              <img
                alt=""
                height="100%"
                src="./790933.jpg"
                width="100%"
              />
            </div>
            <a
              class="carousel__prev"
              href="#carousel__slide2"
            >
              Go to previous slide
            </a>
            <a
              class="carousel__next"
              href="#carousel__slide4"
            >
              Go to next slide
            </a>
          </li>
        </ol>
        <aside
          class="carousel__navigation"
        >
          <ol
            class="carousel__navigation-list"
          >
            <li
              class="carousel__navigation-item"
            >
              <a
                class="carousel__navigation-button"
                href="#carousel__slide1"
              >
                Go to slide 1
              </a>
            </li>
            <li
              class="carousel__navigation-item"
            >
              <a
                class="carousel__navigation-button"
                href="#carousel__slide2"
              >
                Go to slide 2
              </a>
            </li>
            <li
              class="carousel__navigation-item"
            >
              <a
                class="carousel__navigation-button"
                href="#carousel__slide3"
              >
                Go to slide 3
              </a>
            </li>
          </ol>
        </aside>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
